### PR TITLE
[RFC] New supervisor review type

### PR DESCRIPTION
**Pull request by miggol**
In response to #350 - Tag reviews as supervisor reviews properly

Reviews now have a type, creating two mutually exclusive categories of reviews. To me, this makes more sense than a boolean flag, and this allows for future extensibility. Now, supervisor reviews can also make use of the "closed" stage, which should allow us to model all supervisor review states using `review.stage` and `review.go`. So we can now more easily distinguish open and closed supervisor reviews as well. The `continuation` field is unused for supervisor reviews, for now.

This is just meant as a place to propose changes, obviously a lot more plumbing has to happen in the review creation and update process. But we might as well do this right and think ahead.

### CODE REVIEW

File: reviews/models.py [hunk 1]
```diff
@@ -40,6 +40,19 @@ class Review(models.Model):
         (DISCONTINUED, _('Niet verder in behandeling genomen')),
     )
 
+    REVIEW_TYPES = [
+        ("supervisor", _("Beoordeling door eindverantwoordelijke")),
+        ("committee", _("Beoordeling door commissie")),
+    ]
```
Comment (tymees) [inline reviews/models.py:46]: I'd like to see the type values extracted as class constants (like the stages are). That way you can refer to them like this:
``Review.TYPE_SUPERVISOR`` (name is not great, just for illustration purposes).

File: reviews/migrations/0014_auto_20220824_1628.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Generated by Django 3.2.14 on 2022-08-24 14:28
+
+from django.db import migrations
+
+
+def choose_review_type(apps, schema_editor):
```
Comment (tymees) [inline reviews/migrations/0014_auto_20220824_1628.py:6]: This migration should work in theory, but I'm a bit afraid of the 'in practice' part. The DB is full of junk...

Not that you should change anything, I'm just voicing my concern and suggesting we should keep an eye on it when deploying it :P

**Review by tymees — COMMENTED** [review]
Yes, this is definitly a better idea than using a boolean.

So as a course of action, I fully approve.

As a PR, not so much. :P Maybe convert the PR to a Draft PR for clarity?

File: reviews/migrations/0014_auto_20220824_1628.py (hunk 2 above)
Comment (miggol) [reply reviews/migrations/0014_auto_20220824_1628.py]: > This migration should work in theory

If I had the migrations the right way round, then maybe! The order has now been fixed, and `null` is allowed on `review_type` to span the intermediate deadzone.

As an aside, I have now learned that constants like `review.SUPERVISOR` are actually not available in historical models such as those you get in data migrations, unless they are contained in a Meta class. That's why there's a hardcoded 0 there now.

**Comment by miggol** [comment]
Closing as deprecated

**Pull request closed without merging**